### DISCUSSION TITLE
How to monitor CPU/GPU usage in mlflow?

### DISCUSSION THREAD

**Discussion by Jingnan-Jia** [mlflow/mlflow#5992] [Q&A] [answered]
I would like to monitor the CPU/GPU usage duirng machine learning training. Do you know to achieve it in mlflow? I know that some commercial tools like neptune.ai and wand can achieve it but I do not know how to do in in mlflow.

**Comment by sniafas** [comment]
How about a monitoring tool like netdata or just by watching nvidia-smi

**Comment by JHibbard** [comment, marked as answer]
Hi Jingnan-Jia,

MLflow circa 2022 doesn't have built-in tools to help with CPU/GPU monitoring, but if you monitor your resource consumption with an external monitoring tool you can log the information as an artifact or metric. Definitely would be interested in seeing a feature request if you have an idea of how you'd like to see this implemented though.


Hope that helps!
Jim
